### PR TITLE
Adding new script New-StorageAccountTable.ps1 +semver: minor

### PR DESCRIPTION
**Pull request by mwevill**
New-StorageAccountTable.ps1 and Pester Tests.
Automation to Create a Storage account Table Storage.

### CODE REVIEW

File: Infrastructure-Scripts/New-StorageAccountTable.ps1 [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<#
+    .SYNOPSIS
+    Return either the primary or secondary connection string for a Storage Account and write to a Azure Pipelines variable.
```
Comment (adam230594) [inline Infrastructure-Scripts/New-StorageAccountTable.ps1]: Doesn't match with what the script does

File: Infrastructure-Scripts/New-StorageAccountTable.ps1 [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    # --- Check if Storage Account exists in the subscription.
+    $StorageAccountExists = Get-AzStorageAccount -ResourceGroupName $ResourceGroup -Name $StorageAccount -ErrorAction SilentlyContinue
+    if (!$StorageAccountExists) {
+        throw "Storage Account $StorageAccount does not exist."
+    }
+    # --- Create Table Storage.
+    $Key = (Get-AzStorageAccountKey -ResourceGroupName $ResourceGroup -Name $StorageAccount)[0].Value
+    $ctx = New-AzStorageContext -StorageAccountName $StorageAccount -StorageAccountKey $key
+    Get-AzStorageTable -Name $TableName -Context $ctx -ErrorVariable ev -ErrorAction SilentlyContinue
```
Comment (jack-education) [inline Infrastructure-Scripts/New-StorageAccountTable.ps1]:
```suggestion
    Get-AzStorageTable -Name $TableName -Context $ctx -ErrorVariable $ev -ErrorAction SilentlyContinue
```

File: Infrastructure-Scripts/New-StorageAccountTable.ps1 (hunk 1 above)
Comment (adam230594) [inline Infrastructure-Scripts/New-StorageAccountTable.ps1]: Need to update synopsis and description

File: Infrastructure-Scripts/New-StorageAccountTable.ps1 (hunk 2 above)
Comment (adam230594) [reply Infrastructure-Scripts/New-StorageAccountTable.ps1]: I think the original is correct

Comment (adam230594) [reply Infrastructure-Scripts/New-StorageAccountTable.ps1]: EG: https://devblogs.microsoft.com/scripting/handling-errors-the-powershell-way/

Comment (mwevill) [reply Infrastructure-Scripts/New-StorageAccountTable.ps1]: I don't believe that is correct @jack-education  All examples I have ever seen using the variable name not a reference.
In Fact 
  ErrorVariable  String. Specifies a variable that stores errors from
                   the command during processing. This variable is
                   populated in addition to $error.

File: Infrastructure-Scripts/New-StorageAccountTable.ps1 [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    if (!$StorageAccountExists) {
+        throw "Storage Account $StorageAccount does not exist."
+    }
+    # --- Create Table Storage.
+    $Key = (Get-AzStorageAccountKey -ResourceGroupName $ResourceGroup -Name $StorageAccount)[0].Value
+    $ctx = New-AzStorageContext -StorageAccountName $StorageAccount -StorageAccountKey $key
+    $TableExists = Get-AzStorageTable -Name $TableName -Context $ctx  -ErrorAction SilentlyContinue
+    if (!$TableExists) {
+        try {
+            $null = New-AzStorageTable -Name $TableName -Context $ctx
```
Comment (adam230594) [inline Infrastructure-Scripts/New-StorageAccountTable.ps1]: Could you re-add Write-Output ($result), thats nice to have imo

File: Infrastructure-Scripts/New-StorageAccountTable.ps1 [hunk 4]
```diff
@@ -0,0 +1,64 @@
+<#
+    .SYNOPSIS
+    Creates an Azure Storage Table in a Storage Account if there is not one with the supplied Name.cd
```
Comment (dalemorson) [inline Infrastructure-Scripts/New-StorageAccountTable.ps1]:
```suggestion
    Creates an Azure Storage Table in a Storage Account if there is not one with the supplied Name.
```

**Pull request merged**